src/app/Layout.tsx

```diff
@@ -0,0 +1,125 @@
+import { Box, Toolbar, useMediaQuery, useTheme } from '@mui/material';
+import Navigation from 'components/navigation/Navigation';
+import DocsSidePanel from 'components/sidePanelDocs/SidePanel';
+import { NavWidths } from 'context/Theme';
+import { useEffect, useState } from 'react';
+import { ReflexContainer, ReflexElement, ReflexSplitter } from 'react-reflex';
+import { Outlet } from 'react-router';
+import { useLocalStorage } from 'react-use';
+import {
+    useSidePanelDocsStore_animateOpening,
+    useSidePanelDocsStore_setAnimateOpening,
+    useSidePanelDocsStore_show,
+} from 'stores/SidePanelDocs/hooks';
+import { LocalStorageKeys } from 'utils/localStorage-utils';
+
+function AppLayout() {
+    const theme = useTheme();
+    const belowMd = useMediaQuery(theme.breakpoints.down('md'));
+
+    const [navigationConfig, setNavigationConfig] = useLocalStorage(
+        LocalStorageKeys.NAVIGATION_SETTINGS,
+        { open: true }
+    );
+
+    const navigationOpen = navigationConfig?.open ?? true;
+    const navigationWidth: NavWidths = navigationConfig?.open
+        ? NavWidths.FULL
+        : NavWidths.RAIL;
+
+    const toggleNavigationDrawer = () => {
+        setNavigationConfig({ open: !navigationOpen });
+    };
+
+    // Splitter for the side panel docs
+    const [leftPaneFlex, setLeftPaneFlex] = useState<any>(0.0);
+    const [rightPaneFlex, setRightPaneFlex] = useState<any>(0.0);
+    const showDocs = useSidePanelDocsStore_show();
+    const displaySidePanel = showDocs && !belowMd;
```

Comment:
Storing state related to docs. Thinking this _could_ eventually end up in a stand alone hook but not sure yet.